server/cast/utils.go

```diff
@@ -32,6 +32,8 @@ var errOutOfRange = errors.NewKind("%s out of range")
 func handleStringCast(str string, targetType *pgtypes.DoltgresType) (string, error) {
 	tm := targetType.GetAttTypMod()
 	switch targetType.ID {
+	case pgtypes.Text.ID:
+		return str, nil
```

Comment:
I'm not sure why this was added, but I absolutely question it. `handleStringCast` is intended for use with string types that have length limits, so `Text` was intentionally left out. It's unlikely that every "to string" type conversion handles truncation in the same way, so it's very likely that our code is incorrect as it is (we just don't have the testing to really explore it). If `Text` is here too, it just promotes reusing this function rather than diving deeper into casts, which we should be doing. We should remove this.

Reply:
It's there because the cast function for varchar -> text calls it. What's the right way to handle it? Rewrite that cast function?

Reply:
Fixed, this was caused by a bug in the type conversion call